RecoLocalCalo/HGCalRecProducers/plugins/alpaka/HGCalSoALayerClustersProducer.cc

```diff
@@ -0,0 +1,105 @@
+#include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+
+#include "DataFormats/HGCalReco/interface/HGCalSoARecHitsHostCollection.h"
+#include "DataFormats/HGCalReco/interface/alpaka/HGCalSoAClustersDeviceCollection.h"
+#include "DataFormats/HGCalReco/interface/alpaka/HGCalSoARecHitsExtraDeviceCollection.h"
+#include "DataFormats/HGCalReco/interface/HGCalSoAClusters.h"
+#include "RecoLocalCalo/HGCalRecProducers/interface/HGCalSoAClustersExtra.h"
+#include "HGCalLayerClustersSoAAlgoWrapper.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class HGCalSoALayerClustersProducer : public stream::SynchronizingEDProducer<> {
+  public:
+    HGCalSoALayerClustersProducer(edm::ParameterSet const& config)
+        : getTokenDeviceRecHits_{consumes(config.getParameter<edm::InputTag>("hgcalRecHitsSoA"))},
+          getTokenDeviceClusters_{consumes(config.getParameter<edm::InputTag>("hgcalRecHitsLayerClustersSoA"))},
+          deviceTokenSoAClusters_{produces()},
+          thresholdW0_(config.getParameter<double>("thresholdW0")),
+          positionDeltaRho2_(config.getParameter<double>("positionDeltaRho2")) {}
+
+    ~HGCalSoALayerClustersProducer() override = default;
+
+    void acquire(device::Event const& iEvent, device::EventSetup const& iSetup) override {
+      // Get LayerClusters almost-SoA on device: this has still the same
+      // cardinality as the RecHitsSoA, but has all the required information
+      // to assemble the clusters, i.e., it has the cluster index assigned to
+      // each rechit.
+      auto const& deviceInputClusters = iEvent.get(getTokenDeviceClusters_);
+      auto const inputClusters_v = deviceInputClusters.view();
+      //
+      // Allocate output SoA for the clusters, one entry for each cluster
+      auto device_numclusters = cms::alpakatools::make_device_view<const unsigned int>(
+          alpaka::getDev(iEvent.queue()), inputClusters_v.numberOfClustersScalar());
+      auto host_numclusters = cms::alpakatools::make_host_view<unsigned int>(num_clusters_);
+      alpaka::memcpy(iEvent.queue(), host_numclusters, device_numclusters);
+    }
+
+    void produce(device::Event& iEvent, device::EventSetup const& iSetup) override {
+      // Get RecHitsSoA on the device
+      auto const& deviceInputRecHits = iEvent.get(getTokenDeviceRecHits_);
+      auto const inputRechits_v = deviceInputRecHits.view();
+
+      // Get LayerClusters almost-SoA on device: this has still the same
+      // cardinality as the RecHitsSoA, but has all the required information
+      // to assemble the clusters, i.e., it has the cluster index assigned to
+      // each rechit.
+      auto const& deviceInputClusters = iEvent.get(getTokenDeviceClusters_);
+      auto const inputClusters_v = deviceInputClusters.view();
+
+      ALPAKA_ACCELERATOR_NAMESPACE::PortableCollection<HGCalSoAClusters> output(num_clusters_, iEvent.queue());
```

Comment:
would
```suggestion
      PortableCollection<HGCalSoAClusters> output(num_clusters_, iEvent.queue());
```
work ?

We are already inside the `ALPAKA_ACCELERATOR_NAMESPACE`.

Reply:
Or even simpler
```suggestion
      HGCalSoAClustersDeviceCollection output(num_clusters_, iEvent.queue());
```
?